src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlTransformRoutedEvent.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using XamlX.Ast;
+using XamlX.Emit;
+using XamlX.IL;
+using XamlX.Transform;
+using XamlX.TypeSystem;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers;
+
+internal class AvaloniaXamlIlTransformRoutedEvent : IXamlAstTransformer
+{
+    public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode node)
+    {
+        if (node is XamlAstNamePropertyReference prop
+            && prop.TargetType is XamlAstClrTypeReference targetRef
+            && prop.DeclaringType is XamlAstClrTypeReference declaringRef)
+        {
+            var xkt = context.GetAvaloniaTypes();
+            var interactiveType = xkt.Interactivity.Interactive;
+            var routedEventType = xkt.Interactivity.RoutedEvent;
+            var AddHandlerT = xkt.Interactivity.AddHandlerT;
+
+            if (interactiveType.IsAssignableFrom(targetRef.Type))
+            {
+                var eventName = $"{prop.Name}Event";
+                if (declaringRef.Type.GetAllFields().FirstOrDefault(f => f.IsStatic && f.Name == eventName) is { } eventField)
+                {
+                    if (routedEventType.IsAssignableFrom(eventField.FieldType))
+                    {
+                        var instance = new XamlAstClrProperty(prop
+                            , prop.Name
+                            , targetRef.Type
+                            , null
+                            );
+                        instance.Setters.Add(new XamlDirectCallAddHandler(eventField,
+                            targetRef.Type,
+                            xkt.Interactivity.AddHandler,
+                            xkt.Interactivity.RoutedEventHandler
+                            )
+                        );
+                        if (eventField.FieldType.GenericArguments?.Count == 1)
+                        {
+                            var agrument = eventField.FieldType.GenericArguments[0];
+                            if (!agrument.Equals(xkt.Interactivity.RoutedEventArgs))
+                            {
+                                instance.Setters.Add(new XamlDirectCallAddHandler(eventField,
+                                    targetRef.Type,
+                                    xkt.Interactivity.AddHandlerT.MakeGenericMethod([agrument]),
+                                    xkt.EventHandlerT.MakeGenericType(agrument)
+                                    )
+                                );
+                            }
+                        }
+                        return instance;
+                    }
+                    else
+                    {
+                        context.ReportDiagnostic(new XamlX.XamlDiagnostic(
+                            AvaloniaXamlDiagnosticCodes.InvalidXAML,
```

Comment:
@maxkatz6 is this the correct way to handle the error? Do I get a new error code?

Reply:
It's a valid XAML though, so error code is not correct. 

If you want a generic one, then TransformError. (equivalent of throwing XamlTransformException).
But if you want to make it possible to reconfigure this specific error/warning to be ignored (like, with editorconfig), then it needs to have a unique error code.

Reply:
But for this specific error, I believe, TransformError is more than enough.